### PR TITLE
[DOCS] Updates to the runtime fields docs for 7.11+

### LINKED ISSUE #67359: [DOCS] Runtime field updates for 7.11+
This issue tracks the remaining work required for runtime fields documentation in 7.11. 

* Update the [Painless API reference](https://www.elastic.co/guide/en/elasticsearch/painless/current/painless-walkthrough.html) to include mentions of runtime fields, especially the `emit` option in Painless scripts
* Add `runtime_mappings` as a parameter for the request body of the [search API](https://www.elastic.co/guide/en/elasticsearch/reference/master/search-search.html#search-search-api-request-body)
* Update how to [search your data](https://www.elastic.co/guide/en/elasticsearch/reference/master/search-your-data.html) to include searching runtime fields

### PR DESCRIPTION
**Pull request by lockewritesdocs**
This PR:
* Adds a section to _Search your data_ that is specific to [searching data with runtime fields](https://elasticsearch_67484.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/search-your-data.html#run-search-runtime-fields)
* Introduces examples to [Retrieve a runtime field](https://elasticsearch_67484.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/runtime-retrieving-fields.html) that were previously buried in the examples page
* Adds `runtime_mappings` as a parameter for the request body of the [search API](https://elasticsearch_67484.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/search-search.html#search-api-body-runtime)
* Updates the Painless docs to include a new page for [runtime fields](https://elasticsearch_67484.docs-preview.app.elstc.co/guide/en/elasticsearch/painless/master/painless-runtime-fields.html)

Closes #67359

### CODE REVIEW

**Comment by lockewritesdocs** [comment]
@elasticmachine update branch

**Review by javanna — APPROVED** [review]
I left a small comment, LGTM otherwise.

I wonder if we should have more on painless, like more details on the script context for runtime fields. I will leave that to @jdconrad , can also be done in a follow-up if needed

File: docs/reference/search/search-your-data/search-your-data.asciidoc [hunk 1]
```diff
@@ -99,6 +99,72 @@ The API response returns the top 10 documents matching the query in the
 // TESTRESPONSE[s/"took": 5/"took": "$body.took"/]
 // TESTRESPONSE[s/"_id": "kxWFcnMByiguvud1Z8vC"/"_id": "$body.hits.hits.0._id"/]
 
+[discrete]
+[[run-search-runtime-fields]]
+== Define fields that exist only in a query
+Instead of indexing your data and then searching it, you can use define
```
[the remark this answers is not in the document]
Comment (lockewritesdocs) [reply docs/reference/search/search-your-data/search-your-data.asciidoc:105]: Yes, good catch 👏

**Comment by lockewritesdocs** [comment]
> I left a small comment, LGTM otherwise.
> 
> I wonder if we should have more on painless, like more details on the script context for runtime fields. I will leave that to @jdconrad , can also be done in a follow-up if needed

@javanna, are you suggesting that we include a greater explanation of how a runtime field uses the Painless script? Or a deeper discussion of the script formatting?

**Comment by javanna** [comment]
> @javanna, are you suggesting that we include a greater explanation of how a runtime field uses the Painless script? Or adding a specific runtime fields page to Painless contexts?

I meant the latter but I am open to any ideas.

**Review by jdconrad — APPROVED** [review]
LGTM! Thanks for doing this.

@lockewritesdocs As we chatted last week I originally thought adding docs for the Painless context might be overkill, but given the suggestion here from @javanna and how new this feature is, I would rather error on the side of more docs in this case. Maybe this could be added in a follow up PR?

**Comment by lockewritesdocs** [comment]
@jdconrad, sounds good, and your suggestion echoes Luca's sentiments. I'll merge and backport this PR and we can collaborate on a separate PR for adding Painless contexts for runtime fields.

**Pull request merged**